metadata-ingestion/src/datahub/ingestion/graph/filters.py

```diff
@@ -1,16 +1,22 @@
 import dataclasses
 import enum
-from typing import Any, Dict, List, Literal, Optional
+import warnings
+from typing import Dict, List, Literal, Optional, Union
 
 from typing_extensions import TypeAlias
 
 from datahub.emitter.mce_builder import (
     make_data_platform_urn,
     make_dataplatform_instance_urn,
 )
+from datahub.errors import SearchFilterWarning
 from datahub.utilities.urns.urn import guess_entity_type
 
-RawSearchFilterRule = Dict[str, Any]
+RawSearchFilterRule: TypeAlias = Dict[str, Union[str, bool, List[str]]]
+
+# This is a list of OR filters, each of which is a list of AND filters.
```

Comment:
This is commonly referred to as _Disjunctive Normal Form_ (DNF) in boolean logic and computer science: the outer list represents an OR operation, each inner list consists of AND conditions.

We may mention that in the comment or explictely use it in the naming of the type: _eg_ `DisjuntiveNormalFormFilterRule`. Proper type naming is relevant if sdk users are expected to use this class, directly by instantiating the class or as a parameter is some method; not sure if that's the case.

Reply:
this is a public type in our old sdk, so I don't want to rename it fully. The new search sdk does talk about DNF in the internal compile() methods, but I think it's overkill to expose that terminology to the end user